src/grid/TraversabilityGrid.hpp

```diff
@@ -0,0 +1,116 @@
+// DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+// OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+#ifndef __MAPS_TAVERSABILITY_GRID_HPP_
+#define __MAPS_TAVERSABILITY_GRID_HPP_
+
+
+#include "GridMap.hpp"
+#include "TraversabilityCell.hpp"
+#include "TraversabilityClass.hpp"
+
+#include <boost/serialization/access.hpp>
+#include <boost/serialization/nvp.hpp>
+#include <boost/serialization/export.hpp>
+
+namespace maps { namespace grid
+{
+  /**
+   * @brief Maptype to represent the ability of the robot to traverse different 
+   * parts of the terrain.
+   * @details The grid holds TraversabilityCells which in turn hold information 
+   * on the traversability (how well can I drive here?) and probability 
+   * (uncertainty of the data for this point). To conserve storage space both 
+   * values are stored in the grid as an uint8_t. For the traversability this 
+   * represents the Id of a TraversabilityClass in traversabilityClasses which 
+   * holds the actual value. The probability is simply converted from double [0, 1] 
+   * to uint8_t  [0, 255].
+   * */
+  class TraversabilityGrid : public GridMap<TraversabilityCell> 
+  {    
+  public:
+      enum CLASSES {
+            CLASS_UNKNOWN = 0,
+            CLASS_OBSTACLE = 1,
+            CUSTOM_CLASSES = 2
+        };
+	CLASSES classes;
+	
+  private:
+      std::vector<TraversabilityClass> traversabilityClasses;
+          
+  public:
+      TraversabilityGrid();     
+      
+      // Use this if you want a map holding individual data.
+      TraversabilityGrid(const Vector2ui &num_cells, const Vector2d &resolution, const TraversabilityCell default_value);
+      
+      // Use this if the map is supposed to hold data shared with other maps.
+      TraversabilityGrid(const Vector2ui &num_cells, const Vector2d &resolution, const TraversabilityCell default_value, const boost::shared_ptr<LocalMapData> &data);
+    
+      ~TraversabilityGrid();
+      
+      // Set the traversabilityClassId and probability at the given coordinates.
+      // Note that probability has to be in range [0,1].
+      void setTraversabilityAndProbability(uint8_t traversabilityClassId, double probability, size_t x, size_t y);
+      
+      void setTraversability(uint8_t traversabilityClassId, size_t x, size_t y);
+      const TraversabilityClass& getTraversability(size_t x, size_t y) const;
+      
+      // Note that probability has to be in range [0,1].
+      void setProbability(double probability, size_t x, size_t y);
```

Comment:
Since the probability is encoded in a uint8 a float would be enough here as well